app/Http/Controllers/CartController.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Cart;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\JsonResponse;
+
+class CartController extends Controller
+{
+    public function index(): JsonResponse
+    {
+        $cart = Cart::with('items.product')->where('user_id', Auth::id())->first();
+        return response()->json($cart);
+    }
+
+    public function destroy($id): JsonResponse
+    {
+        $cart = Cart::where('user_id', Auth::id())->findOrFail($id);
```

Comment:
what is happening if the user id is not found? maybe you need to have a check bellow and return if the user doesn t exists. Maybe you will get an error on line 23 where you try to delete items from cart